src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp

```diff
@@ -0,0 +1,201 @@
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of Pendulum task. The inverted pendulum swingup problem is a
+ * classic problem in the control literature. In this version of the problem,
+ * the pendulum starts in a random position, and the goal is to swing it up so
+ * it stays upright
+ */
+class Pendulum
+{
+ public:
+  /**
+   * Implementation of state of Pendulum. Each state is a
+   * (theta, angular velocity) vector.
+   */
+  class State
+  {
+   public:
+    /**
+     * Construct a state instance.
+     */
+    State() : data(dimension, arma::fill::zeros)
+    { /* Nothing to do here. */ }
+
+    /**
+     * Construct a state based on the given data.
+     *
+     * @param data Data for the theta and angular velocity.
+     */
+    State(const arma::colvec& data): data(data)
+    { /* Nothing to do here. */ }
+
+    //! Modify the internal representation of the state.
+    arma::colvec& Data() { return data; }
+
+    //! Get the theta.
+    double Theta() const { return data[0]; }
+    //! Modify the value of theta.
+    double& Theta() { return data[0]; }
+
+    //! Get the angular velocity.
+    double AngularVelocity() const { return data[1]; }
+    //! Modify the value of angular velocity.
+    double& AngularVelocity() { return data[1]; }
+
+    //! Encode the state to a column vector.
+    const arma::colvec& Encode() const { return data; }
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 2;
+
+   private:
+    //! Locally-stored (theta, angular velocity) vector
+    arma::colvec data;
+  };
+
+  /**
+   * Implementation of action of Pendulum.
+   * In Pendulum, the action represents the torque to be applied.
+   * This value is bounded in range -2.0 to 2.0 by default.
+   */
+  struct Action
+  {
+    double action[1];
+    // Storing degree of freedom
+    const int size = 1;
+  };
+
+  /**
+   * Construct a Pendulum instance using the given values.
+   *
+   * @param maxAngularVelocity Maximum angular velocity.
+   * @param maxTorque Maximum torque.
+   * @param dt The differential value.
+   */
+  Pendulum(const double maxAngularVelocity = 8,
+           const double maxTorque = 2.0,
+           const double dt = 0.05) :
+      maxAngularVelocity(maxAngularVelocity),
+      maxTorque(maxTorque),
+      dt(dt)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Pendulum. Get reward and next state based 
+   * on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, The reward for taking the action taken for current state.
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState) const
+  {
+    // Get current state
+    double theta = state.Theta();
+    double angular_velocity = state.AngularVelocity();
+
+    // Define required variables
+    double gravity = 10.0;
```

Comment:
Should we use ``constexpr``or ``const`` here to make the expression more clear.

Reply:
@zoq I'm not very knowledgeable about the use of constexpr. After reading here [Difference between `constexpr` and `const`](https://stackoverflow.com/questions/14116003/difference-between-constexpr-and-const) I think that const will be good fit here since these values are not to be modified. What do you suggest ?